### PR TITLE
Inheart 366: Add logout/login

### PR DESCRIPTION
**Pull request by captain-yossarian**
Add logout/login

### CODE REVIEW

File: apps/backend/src/auth/auth.service.ts [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
 export class AuthService {
-  constructor(private readonly jwtService: JwtService) {}
+  constructor(private readonly usersService: UsersService, private readonly jwtService: JwtService) {}
 
   generateJwt(user: User) {
     return this.jwtService.sign({ ...user, typ: 'Bearer' });
   }
+
+  async validateUser(email: string) {
+    const user = await this.usersService.findByEmail(email);
```
Comment (captain-yossarian) [inline apps/backend/src/auth/auth.service.ts:15]: I am not sure why `user` don't have `password` property here

File: libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx [hunk 2]
```diff
@@ -1,12 +1,61 @@
...
+import { useMutation } from '@tanstack/react-query';
 
 import { Button, FormControl, Input } from '@inheartive/ui/atoms';
+import { apiRoutes } from '@inheartive/data';
+import { NativeSyntheticEvent, TextInputChangeEventData } from 'react-native';
+import { useNavigate } from 'react-router-native';
+import { RoutingPath } from '../../../../../../apps/mobile/src/app/routing/routing-path';
+
+export const storage = {
+  access_token: '',
```
Comment (captain-yossarian) [inline libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx:11]: This solution is temporary, I did not figure out yet how to store access tokens in react native. If you know how to handle it please let me know

File: apps/backend/src/auth/auth.service.ts (hunk 1 above)
Comment (KonradKarlinski) [reply apps/backend/src/auth/auth.service.ts:15]: This was playground to improve in the feature.

File: libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx (hunk 2 above)
Comment (KonradKarlinski) [reply libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx:11]: Here are a few options:

AsyncStorage: This is a built-in storage system for React Native that allows you to store key-value pairs. You can use AsyncStorage to store the access token and retrieve it later when needed.

SecureStore: This is a library for securely storing sensitive data in React Native. It uses the native Keychain on iOS and the KeyStore on Android, which are both secure storage systems provided by the operating system.

Redux: If you're using Redux in your React Native app, you can store the access token in the Redux store. This makes it easy to access the token throughout the app, but it also means that the token is stored in memory, which could be a security concern if the app is running on a rooted or jailbroken device.

React context: You can use the React context API to store the access token. This allows you to share data between components without having to pass props down manually through multiple levels of components.

It's important to consider the security requirements of your app when choosing a storage method. If your app handles sensitive data, you should use a secure storage system like SecureStore to protect the access token.

File: apps/backend/src/auth/auth.service.ts (hunk 1 above)
Comment (captain-yossarian) [reply apps/backend/src/auth/auth.service.ts:15]: @KonradKarlinski  So, I need to add a `password` property, right ?

File: libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx (hunk 2 above)
Comment (captain-yossarian) [reply libs/ui/molecules/src/lib/LoginFormControl/LoginFormControl.tsx:11]: @KonradKarlinski thanks, will stick with most secure option

File: apps/mobile/src/app/App.tsx [hunk 3]
```diff
@@ -1,29 +1,36 @@
...
-                  r.needsAuth ? (
-                    <AuthenticatedPageWrapper footerActiveIcon={r.footerIcon}>{r.page}</AuthenticatedPageWrapper>
+                  needsAuth ? (
+                    auth ? (
+                      <AuthenticatedPageWrapper footerActiveIcon={footerIcon}>{page}</AuthenticatedPageWrapper>
+                    ) : (
+                      <SignInPage />
+                    )
                   ) : (
-                    <UnauthenticatedPageWrapper>{r.page}</UnauthenticatedPageWrapper>
```
Comment (captain-yossarian) [inline apps/mobile/src/app/App.tsx:26]: I am not sure we need this component

**Review by KonradKarlinski — APPROVED** [review]

**Pull request merged**